modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -369,26 +357,24 @@ GrainTracker::trackGrains()
 
     if (found_one)
       // Keep track of which new grains the existing ones want to map to
-      new_grain_idx_to_existing_grain_idx[std::make_pair(map_idx, closest_match_idx)].push_back(curr_it->first);
+      new_grain_idx_to_existing_grain_idx[std::make_pair(map_idx, closest_match_idx)].push_back(grain_kv.first);
   }
 
   /**
    * It's possible that multiple existing grains will map to a single new grain (indicated by multiplicity in the
    * new_grain_idx_to_existing_grain_idx data structure).  This will happen any time a grain disappears during
    * this time step. We need to figure out the rightful owner in this case and inactivate the old grain.
    */
-  for (std::map<std::pair<unsigned int, unsigned int>, std::vector<unsigned int> >::iterator it = new_grain_idx_to_existing_grain_idx.begin();
-       it != new_grain_idx_to_existing_grain_idx.end(); ++it)
-  {
-                                                                // map index     feature index
-    MooseSharedPointer<FeatureData> feature_ptr = _feature_sets[it->first.first][it->first.second];
+  for (const auto & new_to_exist_kv : new_grain_idx_to_existing_grain_idx)
+  {                                                             // map index     feature index
+    std::unique_ptr<FeatureData> feature_ptr(std::move(_feature_sets[new_to_exist_kv.first.first][new_to_exist_kv.first.second]));
```

Comment:
I think I would prefer the `=` initialization syntax here... but it's just a preference.

Do you know of any reason to prefer one over the other?